src/main/java/audio/mixer/Mixer.java

```diff
@@ -0,0 +1,60 @@
+package audio.mixer;
+
+import java.util.ArrayList;
+
+public class Mixer {
+    /**
+     * Mixes the given audio streams together.
+     *
+     * @param audioStreams   list of double-arrays
+     * @param startPositions list of start positions for each audio stream (int[0] should always be 0)
+     * @return mixed audio stream as double-array
+     * @throws MixerException if the length of the audio streams does not match
+     */
+    public static double[] mixAudioStreams(ArrayList<double[]> audioStreams, int[] startPositions) throws MixerException {
+
+        // find the longest array
+        int maxLength = 0;
+        for (double[] audioStream : audioStreams) {
+            maxLength = Math.max(maxLength, audioStream.length);
+        }
+        // find the shortest array
+        int minLength = maxLength;
+        for (double[] audioStream : audioStreams) {
+            minLength = Math.min(minLength, audioStream.length);
+        }
+        double[] result = new double[maxLength];
+
+
+        // find the minimum start position (except of startPositions[0] which should always be 0)
+        // check if the start positions are valid
+        int minStartPosition = maxLength;
+        for (int i = 1; i < startPositions.length; i++) {
+            minStartPosition = Math.min(minStartPosition, startPositions[i]);
+            if (startPositions[i] > maxLength) {
+                throw new MixerException("Illegal start position ("+i+"), must be <= maxLength!");
+            }
+        }
```

Comment:
The Sonifier.sonify-method (where the mixer is eventually used) currently does not support AudioStreams of different lengths and introducing such would require major changes (I think) -> the start positions are probably obsolete

Reply:
Alright, then we don't have any problems with overflow at all

Reply:
also there is #81, which should remove our need for individual start-positions